cmd/icap-server/main.go

```diff
@@ -32,21 +29,12 @@ func reqmodHandler(w icap.ResponseWriter, req *icap.Request) {
 			return
 		}
 
-		// If the request is for a CDN URL, delete the Authorization header
-		if cdnRegex.MatchString(req.Request.URL.String()) {
+		// Squid's adaptation_access ACL ensure we receive URLs from cache.allowList.
+		// Only remove Authorization header for content-addressable URLs (containing SHA256).
+		// This prevents breaking caching for arbitrary URLs that need Authorization.
```

Comment:
```suggestion
		// Squid's adaptation_access ACL ensure we receive URLs from cache.allowList.
		// Only remove Authorization header for content-addressable URLs (containing SHA256).
```

The last line is misleading and not even needed. Squid won't cache responses for requests with Authorization headers.

Reply:
updated